src/store/modules/keymap.js

```diff
@@ -442,8 +442,8 @@ const mutations = {
       state.colorwayIndex = index;
     }
   },
-  toggleDisplaySizes(state) {
-    state.displaySizes = !state.displaySizes;
+  setLegends(state, legends) {
```

Comment:
ok, since you're using keymap to store this value, you might as well use keymap to store the legends. I think it's probably better if this were in app tho, since it's not really keymap related, more UI and app. I would validate the values here, something simple like 
```['matrix', 'size','index','keymap'].includes(legends);``` If you keep the values in the keymap store and use them to generate the UI, the UI should never be wrong.